pontoon/machinery/utils.py

```diff
@@ -73,26 +73,34 @@ def get_concordance_search_data(text, locale):
 
     search_query_results = (
         base.models.TranslationMemoryEntry.objects.filter(search_query)
-        .values_list("source", "target", "project__name")
+        .values_list("source", "target", "entity__pk", "project__name", "project__slug")
         .distinct()
     )
 
     search_results = [
         {
             "source": source,
             "target": target,
+            "entity_id": entity_id,
             "project_name": project_name,
+            "project_slug": project_slug,
             "quality": max(
                 int(round(Levenshtein.ratio(text, target) * 100)),
                 int(round(Levenshtein.ratio(text, source) * 100)),
             ),
```

Comment:
We're just adding `quality` to `search_query_results` here... And then we remove it again below.

I wonder if there's a more elegant way to sort, something like:
```python
def key(e):
    quality = max(int(...), int(...))
    return quality, e["target"]

return search_query_results.sort(key=key, reverse=True)
```

Or maybe that's slower? In this case I'd add a comment why we're doing it the way we're doing it.

Reply:
I tried to avoid using too much lambdas and internal functions because they often decrease readability for other people I've worked with. I can switch into your version.